examples/step-87/doc/intro.dox

```diff
@@ -0,0 +1,273 @@
+evaluation. Not surprisingly, such an operation can be also implemented as a cell loop.
+
+<h3>Setup and communication</h3>
+
+To determine the cell $K$ and the reference position $\hat{\boldsymbol x}_q$ within
+the cell for a given point $\boldsymbol{x}_q$ on distributed meshes, deal.II performs
+a two-level-search approach. First, all processes that might possess the point are
+determined ("coarse search"). For this purpose, e.g., a distributed
+tree based on bounding boxes around locally owned domains using "ArborX"
+@cite lebrun2020arborx is applied. After the potentially owning
+processes have been determined and the points have been sent to them as a request,
+one can start to find the cells that surround the points
+among locally owned cells ("fine search").
+In order to accelerate this search, an R-tree from
+"boost::geometry" that is built around the vertices of the mesh is used.
+
+Once the cell $K$ that surrounds point $\boldsymbol{x}_q$ has been found,
+the reference position $\hat{\boldsymbol{x}}_q$ is obtained by performing the minimization:
+@f[
+\min_{\hat{\boldsymbol{x}}_q}(| \boldsymbol{F}_K(\hat{\boldsymbol{x}}_q) - {\boldsymbol{x}_q}|)
+\quad
+\text{with}
+\quad
+\hat{\boldsymbol{x}}_q\in[0,1]^{dim}.
+@f]
+With the determined pieces of information, the desired evaluation can be performed by
+the process that owns the cell. The result can now be communicated to the
+requesting process.
+
+In summary, the coarse search determines, for each point, a list of processes that might own it.
+The subsequent fine search by each process determines whether the processes
+actually own these points by the sequence of request
+("Does the process own the point?") and  answer ("Yes."/"No.").
+Processes might post any number of point requests and communicate with
+any process. We propose to collect the point requests to a process to
+use the dynamic, sparse, scalable consensus-based communication
+algorithms @cite hoefler2010scalable, and to consider the obtained information to
+set up point-to-point communication patterns.
+
+<h3>Implementation: Utilities::MPI::RemotePointEvaluation</h3>
+
+The algorithm described above is implemented in Utilities::MPI::RemotePointEvaluation
+(short: "RPE") and related classes/functions. In this section,
+basic functionalities are briefly summarized. Their advanced
+capabilities will be shown subsequently based on concrete application cases.
+
+The following code snippet shows the setup steps for the communication pattern:
+@code
+std::vector<Point<dim>> points; // ... (filling of points not shown)
+
+RemotePointEvaluation<dim> rpe;
+rpe.reinit(points, triangulation, mapping);
+@endcode
+
+All what is needed is a list of evaluation points and the mesh with a mapping.
+
+The following code snippet shows the evaluation steps:
+@code
+const std::function<void(const ArrayView<T> &, const CellData &)>
+  evaluation_function;
+
+std::vector<T> output;
+rpe.evaluate_and_process(output, evaluation_function);
```

Comment:
I understand your perspective and agree. 

As part of this tutorial, we also include alternative interfaces for the original `evaluate_and_process()` and  `process_and_evaluate()` functions. This eliminates the need to declare the buffer externally, which aligns well with the fact that many use cases do not necessitate buffer reuse. The additional function declarations can be found here
https://github.com/dealii/dealii/blob/63fd25d77a52ae6fb7767717e07cdbca29a2425b/include/deal.II/base/mpi_remote_point_evaluation.h#L206-L211
and here
https://github.com/dealii/dealii/blob/63fd25d77a52ae6fb7767717e07cdbca29a2425b/include/deal.II/base/mpi_remote_point_evaluation.h#L229-L237

From what I can see, another potential improvement could involve modifying the interface of this alternative `evaluate_and_process()` implementation to

```cpp
  template <typename T>
  std::vector<T>
  evaluate_and_process(const std::function<void(const ArrayView<T> &, const CellData &)>
          &evaluation_function) const;
```
which aligns with your suggestion. Do you see any potential drawbacks to this change? It would result in a difference in the number of arguments between `evaluate_and_process()` and `process_and_evaluate()`. However, from my perspective, this appears to be a natural consequence of their different functionalities.